validator/accounts/status.go

```diff
@@ -0,0 +1,176 @@
+package accounts
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"sort"
+
+	"github.com/pkg/errors"
+	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
+	"github.com/prysmaticlabs/prysm/shared/keystore"
+	"go.opencensus.io/trace"
+)
+
+// ValidatorStatusMetadata holds all status information about a validator.
+type ValidatorStatusMetadata struct {
+	PublicKey []byte
+	Metadata  *ethpb.ValidatorStatusResponse
+}
+
+// MaxRequestLimit specifies the max grpc requests allowed
+// to fetch account statuses.
+const MaxRequestLimit = 3
+
+// MaxRequestKeys specifies the max amount of public keys allowed
+// in a single grpc request, when fetching account statuses.
+const MaxRequestKeys = 2000 // XXX: This is an arbitrary number.
```

Comment:
Perhaps in a separate PR this can be configurable through a feature flag, but not important for now. I recommend a higher max request by the way, as we typically use a higher number of keys per validator client at prylabs.